### DISCUSSION TITLE
Getting all SPARQL endpoints from `linkedopendata.eu`

### DISCUSSION THREAD

**Discussion by davidgasquez** [datasets/commons#380] [Show and tell]
The [EU Knowledge Graph](linkedopendata.eu) platform contains structured information about the European Union. They provide an [`SPARQL` endpoint](https://query.linkedopendata.eu/) in which you can run [the following query](https://tinyurl.com/29op92wd) to get all the linked SPARQL endpoints.

```sparql
SELECT ?Endpoint ?EndpointLabel ?URL ?Triples
{?Endpoint wdt:P35 wd:Q3287501;
           wdt:P445 ?URL .
 OPTIONAL {?Endpoint wdt:P579842 ?Triples}
 SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en". }
 
 }

ORDER BY DESC(?Triples)
```

The static results at the time of writing this are available here: [results.csv](https://github.com/datasets/awesome-data/files/11286657/query.csv).

**Comment by davidgasquez** [comment]
Tangential to this, seems is now a good time to revisit the idea of making LLMs understand SPARQL properly. OpenAI recently released some updates to their models and a couple of new features that make working with structured data a lot easier.

I came across a [Wikidata](https://github.com/hardbyte/qabot/blob/main/qabot/functions/wikidata.py) implementation that is able to answer natural language queries like _"How many Hospitals are there located in Beijing"_ by relying on SPARQL and [GPT Functions](https://github.com/hardbyte/qabot/blob/main/qabot/agent.py#L149-L172)!

Would be cool to have a similar approach but directed at building datasets. E.g: "List all hospitals in Beijin with some extra fields like when they were created".

**Reply by rufuspollock** [reply]
@davidgasquez this is really cool - could we turn this into a short blog post. Seems like perfect for a "TIL" or "micropost" section

**Reply by davidgasquez** [reply]
We did [one a while back on using ChatGPT to build SPARKQL Wikidata](https://datahub.io/blog/exporting-sparql-chatgpt). What would be the goal of this new TIL, showing ways to integrate with structured data a bit better? 

I feel there are even better ways since November now, but haven't explored the ecosystem in a while!

**Reply by rufuspollock** [reply]
@davidgasquez still worth the back posting quickly!
